query/src/main/java/org/infinispan/query/CacheQuery.java

```diff
@@ -64,70 +60,28 @@
     */
    CacheQuery<E> maxResults(int numResults);
 
-   /**
-    * @return return the manager for all faceting related operations
-    */
-   FacetManager getFacetManager();
-
    /**
     * Gets the total number of results matching the query, ignoring pagination (firstResult, maxResult).
     *
     * @return total number of results.
     */
-   int getResultSize();
+   long getResultSize();
 
    /**
     * Return the Lucene {@link org.apache.lucene.search.Explanation}
     * object describing the score computation for the matching object/document
     * in the current query
     *
-    * @param documentId Lucene Document id to be explain. This is NOT the object key
+    * @param id The id of the document to test.
     * @return Lucene Explanation
     */
-   Explanation explain(int documentId);
+   Explanation explain(String id);
 
-   /**
-    * Allows lucene to sort the results. Integers are sorted in descending order.
-    *
-    * @param s - lucene sort object
-    */
-   CacheQuery<E> sort(Sort s);
 
-   /**
-    * Defines the Lucene field names projected and returned in a query result
-    * Each field is converted back to it's object representation, an Object[] being returned for each "row"
-    * <p/>
-    * A projectable field must be stored in the Lucene index and use a {@link org.hibernate.search.bridge.TwoWayFieldBridge}
-    * Unless notified in their JavaDoc, all built-in bridges are two-way. All @DocumentId fields are projectable by design.
-    * <p/>
-    * If the projected field is not a projectable field, null is returned in the object[]
-    *
-    * @param fields the projected field names
-    * @return {@code this} to allow for method chaining, but the type parameter now becomes {@code Object[]}
-    */
-   CacheQuery<Object[]> projection(String... fields);
-
-   /**
-    * Enable a given filter by its name.
-    *
-    * @param name of filter.
-    * @return a FullTextFilter object.
-    */
-   FullTextFilter enableFullTextFilter(String name);
-
-   /**
-    * Disable a given filter by its name.
-    *
-    * @param name of filter.
-    */
-   CacheQuery<E> disableFullTextFilter(String name);
-
-   /**
-    * Allows lucene to filter the results.
-    *
-    * @param f - lucene filter
-    */
-   CacheQuery<E> filter(Filter f);
+   // TODO HSEARCH-3325 Restore support for full-text filters
```

Comment:
FYI, we will not support this going forward, all the filtering should be done in the Ickle query itself